app/src/main/java/co/nimblehq/compose/crypto/ui/screens/home/HomeViewModel.kt

```diff
@@ -0,0 +1,55 @@
+package co.nimblehq.compose.crypto.ui.screens.home
+
+import co.nimblehq.compose.crypto.domain.usecase.GetMyCoinsUseCase
+import co.nimblehq.compose.crypto.ui.base.*
+import co.nimblehq.compose.crypto.ui.uimodel.CoinItemUiModel
+import co.nimblehq.compose.crypto.ui.uimodel.toUiModel
+import co.nimblehq.compose.crypto.util.DispatchersProvider
+import dagger.hilt.android.lifecycle.HiltViewModel
+import kotlinx.coroutines.flow.*
+import javax.inject.Inject
+
+private const val MY_COINS_CURRENCY = "usd"
+private const val MY_COINS_ORDER = "market_cap_desc"
+private const val MY_COINS_PRICE_CHANGE_IN_HOUR = "24h"
+private const val MY_COINS_ITEM_PER_PAGE = 10
+
+interface Output : BaseOutput {
+    val myCoins: StateFlow<List<CoinItemUiModel>>
+}
+
+@HiltViewModel
+class HomeViewModel @Inject constructor(
+    dispatchers: DispatchersProvider,
+    private val getMyCoinsUseCase: GetMyCoinsUseCase
+) : BaseViewModel(dispatchers), BaseInput, Output {
+
+    override val input = this
```

Comment:
How about
```suggestion
override val input: BaseInput
        get() = this
```

Reply:
@hoangnguyen92dn my approach `override val input = this` means I will assign this to a new val at initial time. While yours means passing this to val at calling time. They're different, and
- I would like to assign it at the initial time because we don't do to extra operations there.
- `this` is a reference, the value for both cases are the same.

Reply:
@luongvo Yup. But we actually don't use `setter` function, so that's why I suggest to `return this` on the `getter` instead.

Reply:
@Wadeewee will refactor it in the next integration 👌